core/autoscaling.go

```diff
@@ -508,13 +507,12 @@ func (a *autoScalingGroup) havingReadyToAttachSpotInstance() (*string, bool) {
 	return spotInstanceID, false
 }
 
-func (a *autoScalingGroup) allowedInstanceTypes() (string, error) {
+func (a *autoScalingGroup) getAllowedInstanceTypes(baseInstance *instance) string {
```

Comment:
I would return a slice of strings so we don't have to process the result afterwards with `strings.Split`